clang/tools/clang-linker-wrapper/ClangLinkerWrapper.cpp

```diff
@@ -918,8 +851,12 @@ Expected<StringRef> linkDeviceInputFiles(SmallVectorImpl<StringRef> &InputFiles,
 
   SmallVector<StringRef, 8> CmdArgs;
   CmdArgs.push_back(*LLVMLinkPath);
-  for (auto &File : InputFiles)
-    CmdArgs.push_back(File);
+  for (auto &File : InputFiles) {
+    auto IRFile = sycl::convertSPIRVToIR(File, Args);
```

Comment:
I don't really understand this line. Don't we need to check if it's a SPIR-V file? Or can we only hit this with SPV files and not with fat objects?

Reply:
convertSPIRVToIR will handle cases where the file is not a SPIR-V file. it will return the original file if it's not a SPIR-V file.
However, I do think it should be possible to check here if the image type is SPIR-V. i will add a TODO to address this soon. Thanks

Reply:
I would say we rename the function then, the name sounds like it only accepts SPV and always returns IR

Reply:
How about a supporting comment? :-)

Reply:
Added a comment and a TODO to check for SPIR-V image type and exit early on top of function definition.
Thanks